src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -311,10 +311,34 @@ public Commit getLastCommitDetails(Object requestBody) throws VersionControlExce
 
     @Override
     public void createProjectForExercise(ProgrammingExercise programmingExercise) throws VersionControlException {
-        final var exercisePath = programmingExercise.getProjectKey();
-        final var exerciseName = exercisePath + " " + programmingExercise.getTitle();
+        createGitlabGroupForExercise(programmingExercise);
+
+        final Course course = programmingExercise.getCourseViaExerciseGroupOrCourseMember();
+
+        final var instructors = userRepository.getInstructors(course);
+        addUsersToExerciseGroup(instructors, programmingExercise, OWNER);
+
+        // Get editors that are not instructors at the same time
+        final var editors = userRepository.findAllInGroupContainingAndNotIn(course.getEditorGroupName(), new HashSet<>(instructors));
+        addUsersToExerciseGroup(editors, programmingExercise, MAINTAINER);
+
+        // Get teaching assistants that are not instructors nor editors
+        final var instructorsAndEditors = new HashSet<>(instructors);
+        instructorsAndEditors.addAll(editors);
```

Comment:
Just a minor comment: I think it would be more readable if you create an empty hashset and add the instructors in the same way as you do for the editors.

Reply:
Done